x/staking/types/historical_info.go

```diff
@@ -21,8 +23,9 @@ func NewHistoricalInfo(header tmproto.Header, valSet Validators) HistoricalInfo
 }
 
 // MustMarshalHistoricalInfo wll marshal historical info and panic on error
-func MustMarshalHistoricalInfo(cdc codec.BinaryMarshaler, hi HistoricalInfo) []byte {
-	return cdc.MustMarshalBinaryBare(&hi)
+// TODO: why don't we use codec.LegacyAmino?
```

Comment:
It's used here:
https://github.com/cosmos/cosmos-sdk/blob/505d424a3bc33e089a0809494e11ff49020b54a2/x/staking/keeper/historical_info.go#L28-L29

and `k.cdc` (which comes from e.g. `simapp.appCodec`) can be either proto or amino.